src/inc/msquichelper.h

```diff
@@ -494,4 +509,101 @@ FreeServerConfiguration(
     MsQuic->ConfigurationClose(Configuration);
 }
 
+inline
+void
+WriteSslKeyLogFile(
+    _In_z_ const char* FileName,
+    _In_ QUIC_TLS_SECRETS& TlsSecrets
+    )
+{
+    FILE* File = nullptr;
+#ifdef _WIN32
+    if (fopen_s(&File, FileName, "ab")) {
+        printf("Failed to open sslkeylogfile %s\n", FileName);
+        return;
+    }
+#else
+    File = fopen(FileName, "ab");
+#endif
+
+    if (File == nullptr) {
+        printf("Failed to open sslkeylogfile %s\n", FileName);
+        return;
+    }
+    if (fseek(File, 0, SEEK_END) == 0 && ftell(File) == 0) {
+        fprintf(File, "# TLS 1.3 secrets log file, generated by quicinterop\n");
+    }
+    char ClientRandomBuffer[(2 * sizeof(QUIC_TLS_SECRETS::ClientRandom)) + 1] = {0};
+    char TempHexBuffer[(2 * QUIC_TLS_SECRETS_MAX_SECRET_LEN) + 1] = {0};
+    if (TlsSecrets.IsSet.ClientRandom) {
```

Comment:
Does it make sense to do anything in this function at all if `ClientRandom` isn't set?

Reply:
Well, you can manually populate the CLIENT_RANDOM if you're looking at a wireshark capture of only one connection.